lib/setopt.c

```diff
@@ -1606,6 +1652,13 @@ CURLcode Curl_vsetopt(struct Curl_easy *data, CURLoption option, va_list param)
     result = Curl_setstropt(&data->set.str[STRING_CERT_ORIG],
                             va_arg(param, char *));
     break;
+  case CURLOPT_SSLCERT_BLOB:
+    /*
+     * String that holds file name of the SSL certificate to use
+     */
+    result = Curl_setblobopt(&data->set.str[STRING_CERT_ORIG],
```

Comment:
The blob is not a string, I don't like this abuse of the string array.

Reply:
On VTLS, we have code like
curl_decode_data_blob(data->set.ssl.cert, ....)

Do you prefer , on struct ssl_config_data in urldata.h adding 
after
  char *cert; /* client certificate file name */
adding
  struct curl_blob *cert_blob; /* NEW : client certificate blob */

after
  char *key; /* private key file name */
adding
  struct curl_blob *key_blob; /* NEW : private key blob */

so we need no magic in blob

Reply:
I rewrote the blob stuff

I made a branch https://github.com/gvollant/curl/tree/pkcs12mem/pcks12memstringblob with my previous code (with magic).

Now, I fully removed magic, and never share variable between string filename and blob. Less confusion, less complex.

They are probably some comment or variable name to update.
Only PKCS12 on openssl and Schannel currently work. 
I'll do other cert in openssl and MacOSX securetransport work when we will be OK on the blob design

Reply:
the updated code is at https://github.com/gvollant/curl/tree/pkcs12mem/pkcs12mem and link to this PR